genesis/genesis_test.go

```diff
@@ -32,14 +36,316 @@ func TestAliases(t *testing.T) {
+			networkID: 12345,
+			config: func() *Config {
+				thisConfig := LocalConfig
+				thisConfig.StartTime = 999999999999999
+				return &thisConfig
+			}(),
+			err: "start time cannot be in the future",
+		},
+		"no initial supply": {
+			networkID: 12345,
+			config: func() *Config {
+				thisConfig := LocalConfig
+				thisConfig.Allocations = []Allocation{}
+				return &thisConfig
+			}(),
+			err: "initial supply must be > 0",
+		},
+		"no initial stakers": {
+			networkID: 12345,
+			config: func() *Config {
+				thisConfig := LocalConfig
+				thisConfig.InitialStakers = []Staker{}
+				return &thisConfig
+			}(),
+			err: "initial stakers must be > 0",
+		},
+		"invalid initial stake duration": {
+			networkID: 12345,
+			config: func() *Config {
+				thisConfig := LocalConfig
+				thisConfig.InitialStakeDuration = 0
+				return &thisConfig
+			}(),
+			err: "initial stake duration must be > 0",
+		},
+		"invalid stake offset": {
+			networkID: 12345,
+			config: func() *Config {
+				thisConfig := LocalConfig
+				thisConfig.InitialStakeDurationOffset = 100000000
+				return &thisConfig
+			}(),
+			err: "initial stake duration is 31536000 but need at least 400000000 with offset of 100000000",
+		},
+		"empty initial staked funds": {
+			networkID: 12345,
+			config: func() *Config {
+				thisConfig := LocalConfig
+				thisConfig.InitialStakedFunds = []ids.ShortID(nil)
+				return &thisConfig
+			}(),
+			err: "initial staked funds cannot be empty",
+		},
+		"duplicate initial staked funds": {
+			networkID: 12345,
+			config: func() *Config {
+				thisConfig := LocalConfig
+				thisConfig.InitialStakedFunds = append(thisConfig.InitialStakedFunds, thisConfig.InitialStakedFunds[0])
+				return &thisConfig
+			}(),
+			err: "duplicated in initial staked funds",
+		},
+		"initial staked funds not in allocations": {
+			networkID: 5,
+			config: func() *Config {
+				thisConfig := FujiConfig
+				thisConfig.InitialStakedFunds = append(thisConfig.InitialStakedFunds, LocalConfig.InitialStakedFunds[0])
+				return &thisConfig
+			}(),
+			err: "does not have an allocation to stake",
+		},
+		"empty C-Chain genesis": {
+			networkID: 12345,
+			config: func() *Config {
+				thisConfig := LocalConfig
+				thisConfig.CChainGenesis = ""
+				return &thisConfig
+			}(),
+			err: "C-Chain genesis cannot be empty",
+		},
+		"empty message": {
+			networkID: 12345,
+			config: func() *Config {
+				thisConfig := LocalConfig
+				thisConfig.Message = ""
+				return &thisConfig
+			}(),
+			err: "genesis message cannot be empty",
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			err := validateConfig(test.networkID, test.config)
+			if len(test.err) > 0 {
+				if !strings.Contains(err.Error(), test.err) {
+					t.Fatalf(`expected error containing "%s" but got "%s"`,
+						test.err,
+						err.Error(),
+					)
+				}
+				return
+			}
+			if len(test.err) == 0 && err != nil {
+				t.Fatal(err)
```

Comment:
We're trying to use the `assert` package in new tests...would you mind using it in this PR?
e.g.
`assert.FailNow(t,err)` instead of `t.Fatalf(err)`

Reply:
Gotcha! Was following the pattern in this file to use `testing`...will migrate.

Reply:
Ty! Yeah, switching to `assert` is pretty recent. It just makes things a little more readable